python/cudf_polars/cudf_polars/utils/config.py

```diff
@@ -5,209 +5,122 @@
-        ----------
-        name
-            Period-separated config name.
-        value
-            New config value.
-        """
-        options = config_options = copy.deepcopy(self.config_options)
-        keys = name.split(".")
-        for k in keys[:-1]:
-            assert isinstance(options, dict)
-            if k not in options:
-                options[k] = {}
-            options = options[k]
-        options[keys[-1]] = value
-        return type(self)(config_options)
-
-    def get(self, name: str, *, default: Any = None) -> Any:
-        """
-        Get a user config option.
+@dataclasses.dataclass(frozen=True, eq=True)
+class StreamingExecutor:
+    """
+    Configuration for the cudf-polars streaming executor.
 
-        Nested dictionary keys should be separated by periods.
-        For example::
+    Parameters
+    ----------
+    scheduler
+        The scheduler to use for the streaming executor.
 
-            >>> chunked = config_options.get("parquet_options.chunked")
+        "distributed" requires a Dask cluster to be running.
+    """
 
-        Parameters
-        ----------
-        name
-            Period-separated config name.
-        default
-            Default return value.
+    name: Literal["streaming"] = "streaming"
+    scheduler: Literal["synchronous", "distributed"] = "synchronous"
+    fallback_mode: FALLBACK_MODE = "warn"
+    max_rows_per_partition: int = 1_000_000
+    cardinality_factor: dict[str, float] = dataclasses.field(default_factory=dict)
+    parquet_blocksize: int = 1_000_000_000  # why isn't this a ParquetOption?
+    groupby_n_ary: int = 32
+    broadcast_join_limit: int = 4
+    shuffle_method: Literal["tasks"] | None = None
 
-        Returns
-        -------
-        The user-specified config value, or `default`
-        if the config is not found.
-        """
-        options = self.config_options
-        keys = name.split(".")
-        for k in keys[:-1]:
-            assert isinstance(options, dict)
-            options = options.get(k, {})
-        return options.get(keys[-1], default)
-
-    def __hash__(self) -> int:
-        """Hash a ConfigOptions object."""
-        try:
-            return self._hash_value
-        except AttributeError:
-            self._hash_value = hash(json.dumps(self.config_options))
-            return self._hash_value
-
-    @staticmethod
-    def validate(config: dict) -> dict:
-        """
-        Validate a configuration-option dictionary.
 
-        Parameters
-        ----------
-        config
-            GPUEngine configuration options to validate.
+@dataclasses.dataclass(frozen=True, eq=True)
+class InMemoryExecutor:
+    """Configuration for the cudf-polars in-memory executor."""
 
-        Returns
-        -------
-        Valid config-option dictionary.
+    name: Literal["in-memory"] = "in-memory"
+    scheduler: Literal["synchronous"] = "synchronous"
+    shuffle_method: Literal["tasks", "rapidsmpf"] | None = None
```

Comment:
I think I mixed this up while adapting https://github.com/rapidsai/cudf/blob/76baf5ccad27f0858347e1ed8d2b6acd5f114b64/python/cudf_polars/cudf_polars/utils/config.py#L204-L210. The InMemoryExecutor should only allow `tasks` IIUC.